src/templates/blog-post.js

```diff
@@ -1,42 +1,65 @@
 import React from 'react'
-import { graphql } from 'gatsby'
-import { Helmet } from 'react-helmet'
+import { Link, graphql } from 'gatsby'
 import get from 'lodash/get'
-import Img from 'gatsby-image'
-import Layout from '../components/layout'
 
-import heroStyles from '../components/hero.module.css'
+import Seo from '../components/seo'
+import Layout from '../components/layout'
+import Hero from '../components/hero'
+import Tags from '../components/tags'
+import * as styles from './blog-post.module.css'
 
 class BlogPostTemplate extends React.Component {
   render() {
     const post = get(this.props, 'data.contentfulBlogPost')
-    const siteTitle = get(this.props, 'data.site.siteMetadata.title')
+    const previous = get(this.props, 'data.previous')
+    const next = get(this.props, 'data.next')
 
     return (
       <Layout location={this.props.location}>
-        <div style={{ background: '#fff' }}>
-          <Helmet title={`${post.title} | ${siteTitle}`} />
-          <div className={heroStyles.hero}>
-            <Img
-              className={heroStyles.heroImage}
-              alt={post.title}
-              fluid={post.heroImage.fluid}
-            />
-          </div>
-          <div className="wrapper">
-            <h1 className="section-headline">{post.title}</h1>
-            <p
-              style={{
-                display: 'block',
-              }}
-            >
-              {post.publishDate}
-            </p>
+        <Seo
+          title={post.title}
+          description={post.description.childMarkdownRemark.excerpt}
+          image={`http:${post.heroImage.resize.src}`}
+        />
+        <Hero
+          image={post.heroImage.gatsbyImageData}
+          title={post.title}
+          content={post.description.childMarkdownRemark.excerpt}
+        />
+        <div className={styles.container}>
+          <span className={styles.meta}>
+            {post.author.name} &middot;{' '}
+            <time dateTime={post.rawDate}>{post.publishDate}</time> –{' '}
+            {post.body.childMarkdownRemark.timeToRead} minute read
+          </span>
+          <div className={styles.article}>
             <div
+              className={styles.body}
               dangerouslySetInnerHTML={{
                 __html: post.body.childMarkdownRemark.html,
               }}
             />
+            <Tags tags={post.tags} />
+            {(previous || next) && (
+              <nav>
+                <ul className={styles.articleNavigation}>
+                  <li>
+                    {previous && (
```

Comment:
Do we want to make it where the `li` itself doesn't render if there is no `previous` or `next`?

Reply:
Good catch. We should add the check up one level otherwise we'll render (I think?) empty li.

It's possible the spacing is useful / needed to position in the correct spot, but I haven't pulled it down locally so purely speculative :)

Reply:
(and there are better ways to do this than using empty elements, of course)

Reply:
:+1: Agreed that we should catch this (and [that](https://github.com/contentful/starter-gatsby-blog/pull/168#discussion_r677787280)), will adjust. :v:

Reply:
Fixed in bc2f69b.